test/fake_clock.h

```diff
@@ -51,6 +51,20 @@ class FakeClock {
 }  // namespace testing
 }  // namespace google
 
+namespace boost {
+namespace asio {
+// Specialize boost::asio::wait_traits so that
+// basic_waitable_timer::async_wait() will work with a FakeClock.
+template<>
+struct wait_traits<google::testing::FakeClock> {
+  static google::testing::FakeClock::duration to_wait_duration(
+      const google::testing::FakeClock::duration& d) {
+    return std::min(d, google::time::seconds(0.001));
```

Comment:
Is this correct? Should the wait be at *most* 1ms? Or did you mean at *least* 1ms?

Reply:
Yep, it should be at most.

FYI, I saw this idiom a few places; eg the "Jumping Through Time" section of http://blog.think-async.com/2007/08/time-travel.html.

Reply:
Done.

Note that to_posix_duration is for a deadline_timer, whereas to_wait_duration is for a basic_waitable_timer. I added this to the comment.